### ISSUE TITLE
Intermittent - "The Document could not be saved"

### ISSUE THREAD

**Issue by dhumes0524** [mstarke/MacPass#545]
Hello, I am experiencing an intermittent issue where when pressing Save, I get the alert "The Document {docname} could not be saved".  I can then press Save again, and it will work.  Approximately 80% of the time when I press save it does not work, but sometimes it randomly will. 
Here is my configuration:

MacPass 0.6.2
Using Password and Keyfile
Keyfile is located on a Microsoft OneDrive folder that I am syncing to my Mac via OneDrive app.
Database (.kdbx) is located on a mounted (AFP) network file share. 
No known connectivity or link issues to the network.  Ping test to file share has no dropped packets.
Database contains an entry in the top root-level General folder.  (saw this recommended on another thread)
Mac is running macOS 10.12.3

Thoughts?

Thank you!

**Comment by mstarke** [comment]
Can you take a look at the log (Console.app) to see if anything more substantial gets written there. I'll try to extend the loggin for writing to see why this might happen.

**Comment by dhumes0524** [comment]
I am not very skilled on reviewing the console, but when I try the save it shows 4 listings for ‘macpass’
[cid:image001.png@01D27BA2.00E59510]

And the only one that has any details to it is the highlighted one, and here are the details: [cid:image002.png@01D27BA2.00E59510]

**Comment by mstarke** [comment]
It seems the attached images got lost. You might need to use the Github web site to upload them.

**Comment by dhumes0524** [comment]
Yes that appears so.  Also, for whatever reason, my console seems to be reporting differently this time.  Attached are the results inside of console from the time I open my database (9:14AM) through trying to save multiple times and failing (last one at 9:17AM)

[ConsoleLog copy.txt](https://github.com/mstarke/MacPass/files/742152/ConsoleLog.copy.txt)

**Comment by mstarke** [comment]
Can you give me the build number of MacPass (it's listed in the about box in parenthesis after the version).

**Comment by dhumes0524** [comment]
Build 12420

**Comment by dhumes0524** [comment]
Any other information I can get you?

**Comment by mstarke** [comment]
Not really, I'll need to take a look into this but it seems that MacPass actually loses the file connection. What bothers me is the read-only preferences. Do you use some net mounted user directory?

**Comment by dhumes0524** [comment]
My user directory is local.  The location of the database is a mounted network share.  Also the location of the keyfile is a mounted OneDrive folder.

**Comment by dhumes0524** [comment]
Moving my database off my my AFP mounted network share fixes the issue and it saves everytime, FYI.  This isn’t ideal though because I want to have my database shared so that other members of the IT Dept are able to reach it.

**Comment by mstarke** [comment]
Thank you for the feedback. I'll try to replicate an AFP share and see how to get past those bugs.
